cmd/alias.go

```diff
@@ -0,0 +1,153 @@
+		return 1
+	}
+
+	args = c.flags.Args()
+
+	switch len(args) {
+	case 0:
+	default:
+		c.UI.Error(fmt.Sprintf("Error: too many arguments (expected 0, got %d)\n", len(args)))
+		c.UI.Output(c.Help())
+		return 1
+	}
+
+	// Copy files from package to Trellis
+	files := []string{"alias.yml", "alias.yml.j2", "alias-copy.yml"}
+	for _, file := range files {
+		_, filename, _, _ := runtime.Caller(0)
+		source := path.Join(path.Dir(filename), "../playbooks/" + file)
+		copyPlaybook(source, file)
+		defer deletePlaybook(file)
+	}
+
+	environments := c.Trellis.EnvironmentNames()
+	var remoteEnvironments []string
+	for _, environment := range environments {
+		if environment != c.local {
+			remoteEnvironments = append(remoteEnvironments, environment)
+		}
+	}
+
+	tempDir, tempDirErr := ioutil.TempDir("", "trellis-alias-")
+	if tempDirErr != nil {
+		log.Fatal(tempDirErr)
+	}
+	defer os.RemoveAll(tempDir)
+
+	for _, environment := range remoteEnvironments {
+		alias := execCommand("ansible-playbook", "alias.yml", "-vvv","-e", "env=" + environment, "-e", "trellis_alias_j2=alias.yml.j2", "-e", "trellis_alias_temp_dir=" + tempDir)
+		appendEnvironmentVariable(alias, "ANSIBLE_RETRY_FILES_ENABLED=false")
+
+		logCmd(alias, c.UI, true)
+		err := alias.Run()
+
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error running ansible-playbook alias.yml: %s", err))
+			return 1
+		}
+	}
+
+	combined := ""
+	for _, environment := range remoteEnvironments {
+		part, err := ioutil.ReadFile(tempDir + "/" + environment + ".yml.part")
+		if err != nil {
+			log.Fatal(err)
+		}
+		combined = combined + string(part)
+	}
+
+	writeFileErr := ioutil.WriteFile(tempDir + "/combined.yml", []byte(combined), 0644)
+	if writeFileErr != nil {
+		log.Fatal(writeFileErr)
+	}
+
+	aliasCopy := execCommand("ansible-playbook", "alias-copy.yml", "-e", "env=" + c.local, "-e", "trellis_alias_combined=" + tempDir + "/combined.yml")
+	appendEnvironmentVariable(aliasCopy, "ANSIBLE_RETRY_FILES_ENABLED=false")
+
+	logCmd(aliasCopy, c.UI, true)
+	aliasCopyErr := aliasCopy.Run()
+
+	if aliasCopyErr != nil {
+		c.UI.Error(fmt.Sprintf("Error running ansible-playbook alias-copy.yml: %s", aliasCopyErr))
+		return 1
+	}
+
+	c.UI.Info(color.GreenString("✓ wp-cli.trellis-alias.yml generated"))
+	message := `
+Action Required: Add these lines into wp-cli.yml or wp-cli.local.yml
+
+_: 
+  inherit: wp-cli.trellis-alias.yml
+`
+	c.UI.Info(strings.TrimSpace(message))
+
+	return 0
+}
+
+func (c *AliasCommand) Synopsis() string {
+	return "Generate WP CLI aliases for remote environments"
+}
+
+func (c *AliasCommand) Help() string {
+	helpText := `
+Usage: trellis alias [options]
+
+Generate WP CLI aliases for remote environments
+
+Options:
+      --local (default: development) Local environment name
+  -h, --help  show this help
+`
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *AliasCommand) AutocompleteFlags() complete.Flags {
+	return complete.Flags{
+		"--local": complete.PredictNothing,
```

Comment:
We could autocomplete environment names. `c.Trellis.AutocompleteEnvironment()` may just work here instead

Reply:
Added in nhttps://github.com/roots/trellis-cli/pull/52/commits/e49ea2126c7062a1006060730c9d737bf7c6becf, but not sure how to verify it is workinng

Reply:
I test it out locally by adding the build binary to `$PATH` and manually installing the shell completions (not on that laptop so I forget the cmd, but I also use zsh).

Reply:
Added it into my `$PATH`

```
➜ trellis-cli --autocomplete-install
2019/08/20 17:45:14 1 error occurred:
	* already installed in /Users/me/.zshrc
```

Verified `.zshrc` has:

```
complete -o nospace -C /Users/tangrufus/Code/trellis-cli/bin/trellis-cli trellis
```

However, no shell completions avaiable for me.
```
➜ trellis-cli alias --local=
```

Note: There could be a chance that my local zsh setup is incorrect.

Reply:
And that's from within a valid Trellis project? Oh you know what... that completion may only work for positional args. I don't think any other commands use it for flags yet so that's my guess

Reply:
Reverted to `PredictNothing`.

However, we found another issue: `runtime.Caller` doesn't work when installing via `brew install --HEAD XXX/tap/trellis-cli-dev`. To be investigated.